// @founding/devkit/dist/index.js
var $12jGn$crossspawn = require("cross-spawn");
var $12jGn$chalk = require("chalk");
var $12jGn$consoletableprinter = require("console-table-printer");
var $12jGn$ora = require("ora");
var $12jGn$readline = require("readline");
var $12jGn$prompts = require("prompts");
var $12jGn$prettier = require("prettier");
var $12jGn$ejs = require("ejs");
var $12jGn$jscodeshift = require("jscodeshift");
var $12jGn$mrleeboprismaast = require("@mrleebo/prisma-ast");
var $12jGn$fsextra = require("fs-extra");
var $12jGn$globby = require("globby");
var $12jGn$rimraf = require("rimraf");
var $12jGn$updatecheck = require("update-check");

function $parcel$exportWildcard(dest, source) {
  Object.keys(source).forEach(function(key) {
    if (key === 'default' || key === '__esModule' || dest.hasOwnProperty(key)) {
      return;
    }

    Object.defineProperty(dest, key, {
      enumerable: true,
      get: function get() {
        return source[key];
      }
    });
  });

  return dest;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
var $40b2ad338fe99ac0$exports = {};

$parcel$export($40b2ad338fe99ac0$exports, "filterDependencies", () => $40b2ad338fe99ac0$export$a48cded481a7b9a3);
$parcel$export($40b2ad338fe99ac0$exports, "filterTemplates", () => $40b2ad338fe99ac0$export$883cc5315f212021);
const $40b2ad338fe99ac0$var$filterList = (list, context)=>{
    return list.filter(({ filters: filters  })=>{
        if (filters) return filters.every((filter)=>filter(context)
        );
        return true;
    });
};
const $40b2ad338fe99ac0$export$a48cded481a7b9a3 = (dependencies, context)=>{
    return $40b2ad338fe99ac0$var$filterList(dependencies, context);
};
const $40b2ad338fe99ac0$export$883cc5315f212021 = (templates, context)=>{
    return $40b2ad338fe99ac0$var$filterList(templates, context);
};


var $a272df753f76392f$exports = {};

$parcel$export($a272df753f76392f$exports, "startDocker", () => $a272df753f76392f$export$b9c77372957d1829);
$parcel$export($a272df753f76392f$exports, "stopDocker", () => $a272df753f76392f$export$9238c7a74ca8f573);
$parcel$export($a272df753f76392f$exports, "runMigrations", () => $a272df753f76392f$export$1fe1303d5ebdc741);
$parcel$export($a272df753f76392f$exports, "runPrismaCodegen", () => $a272df753f76392f$export$85f637821bb9f000);
$parcel$export($a272df753f76392f$exports, "syncGeneratorMigrations", () => $a272df753f76392f$export$d1a4d70edc68db32);
var $ba05d75a0e232cea$exports = {};

$parcel$export($ba05d75a0e232cea$exports, "execSync", () => $ba05d75a0e232cea$export$fa77af024d7e00a8);
$parcel$export($ba05d75a0e232cea$exports, "exec", () => $ba05d75a0e232cea$export$78e3044358792147);
$parcel$export($ba05d75a0e232cea$exports, "sleep", () => $ba05d75a0e232cea$export$e772c8ff12451969);

const $ba05d75a0e232cea$export$fa77af024d7e00a8 = (command, args, options = {
})=>{
    return ($parcel$interopDefault($12jGn$crossspawn)).sync(command, args, options);
};
const $ba05d75a0e232cea$export$78e3044358792147 = (command, args, withStdio = true)=>new Promise((resolve, reject)=>{
        const child = ($parcel$interopDefault($12jGn$crossspawn))(command, args, {
            stdio: withStdio ? "inherit" : "ignore",
            env: {
                ...process.env,
                ADBLOCK: "1",
                DISABLE_OPENCOLLECTIVE: "1",
                TS_NODE_COMPILER_OPTIONS: JSON.stringify({
                    module: "commonjs"
                })
            }
        });
        child.on("close", (code)=>{
            if (code !== 0) {
                reject({
                    command: `${command} ${args.join(" ")}`
                });
                return;
            }
            resolve();
        });
    })
;
const $ba05d75a0e232cea$export$e772c8ff12451969 = (ms)=>{
    return new Promise((resolve)=>setTimeout(resolve, ms)
    );
};


var $e7da8e8b58079630$exports = {};

$parcel$export($e7da8e8b58079630$exports, "table", () => $e7da8e8b58079630$export$9852986a3ec5f6a0);
$parcel$export($e7da8e8b58079630$exports, "chalk", () => $e7da8e8b58079630$export$a0b6f61a94abd245);
$parcel$export($e7da8e8b58079630$exports, "logger", () => $e7da8e8b58079630$export$af88d00dbe7f521);




const $e7da8e8b58079630$export$9852986a3ec5f6a0 = $12jGn$consoletableprinter.Table;
const $e7da8e8b58079630$export$a0b6f61a94abd245 = ($parcel$interopDefault($12jGn$chalk));
const $e7da8e8b58079630$export$af88d00dbe7f521 = {
    withWarning (str) {
        return `⚠️  ${($parcel$interopDefault($12jGn$chalk)).yellow(str)}`;
    },
    withCaret (str) {
        return `${($parcel$interopDefault($12jGn$chalk)).gray(">")} ${str}`;
    },
    withCheck (str) {
        return `${($parcel$interopDefault($12jGn$chalk)).green("✔")} ${str}`;
    },
    withX (str) {
        return `${($parcel$interopDefault($12jGn$chalk)).red.bold("✕")} ${str}`;
    },
    withProgress (str) {
        return $e7da8e8b58079630$export$af88d00dbe7f521.withCaret(str);
    },
    withError (str) {
        return $e7da8e8b58079630$export$af88d00dbe7f521.withX(($parcel$interopDefault($12jGn$chalk)).red.bold(str));
    },
    withMeta (str) {
        return ($parcel$interopDefault($12jGn$chalk)).dim(str);
    },
    withVariable (str) {
        return ($parcel$interopDefault($12jGn$chalk)).green(`${str}`);
    },
    withCommand (str) {
        return ($parcel$interopDefault($12jGn$chalk)).cyan(`${str}`);
    },
    withIndent (str, indent = 0) {
        const indents = "   ".repeat(indent);
        return `${indents}${str}`;
    },
    spinner (str) {
        const spinner = ($parcel$interopDefault($12jGn$ora))({
            text: str,
            color: "cyan",
            spinner: "dots"
        });
        if (str) spinner.start();
        return spinner;
    },
    clearLine (msg) {
        ($parcel$interopDefault($12jGn$readline)).clearLine(process.stdout, 0);
        ($parcel$interopDefault($12jGn$readline)).cursorTo(process.stdout, 0);
        msg && process.stdout.write(msg);
    },
    clearConsole () {
        if (process.platform === "win32") process.stdout.write("\x1B[2J\x1B[0f");
        else process.stdout.write("\x1B[2J\x1B[3J\x1B[H");
    },
    title (msg) {
        console.log(($parcel$interopDefault($12jGn$chalk)).bold(msg));
    },
    list (msgs) {
        console.log(msgs.map((m)=>$e7da8e8b58079630$export$af88d00dbe7f521.withCaret(m)
        ).join("\n"));
    },
    error (msg) {
        console.error($e7da8e8b58079630$export$af88d00dbe7f521.withX(($parcel$interopDefault($12jGn$chalk)).red.bold(msg)));
    },
    warning (msg) {
        console.log($e7da8e8b58079630$export$af88d00dbe7f521.withWarning(msg));
    },
    meta (msg) {
        console.log($e7da8e8b58079630$export$af88d00dbe7f521.withMeta(msg));
    },
    progress (msg) {
        console.log($e7da8e8b58079630$export$af88d00dbe7f521.withCaret(msg));
    },
    success (msg) {
        console.log($e7da8e8b58079630$export$af88d00dbe7f521.withCheck(msg));
    },
    log (msg, indents = 0, newLine = false) {
        console.log($e7da8e8b58079630$export$af88d00dbe7f521.withIndent(msg, indents));
        if (newLine) console.log();
    },
    newLine () {
        console.log();
    },
    table: $e7da8e8b58079630$export$9852986a3ec5f6a0
};


const $a272df753f76392f$export$b9c77372957d1829 = async ()=>{
    console.log("Starting docker instance");
    await $ba05d75a0e232cea$export$78e3044358792147("docker-compose", [
        "up",
        "-d"
    ]);
};
const $a272df753f76392f$export$9238c7a74ca8f573 = async ()=>{
    console.log("Stopping docker instance");
    await $ba05d75a0e232cea$export$78e3044358792147("docker-compose", [
        "down"
    ]);
};
const $a272df753f76392f$export$1fe1303d5ebdc741 = async (name = "")=>{
    console.log("Running local migrations");
    await $ba05d75a0e232cea$export$78e3044358792147("npm", [
        "run",
        "prisma:migrate:dev",
        "--",
        "-n",
        name,
        "--skip-seed", 
    ]);
};
const $a272df753f76392f$export$85f637821bb9f000 = async ()=>{
    console.log("Running prisma codegen");
    await $ba05d75a0e232cea$export$78e3044358792147("npm", [
        "run",
        "prisma:generate"
    ]);
};
const $a272df753f76392f$export$d1a4d70edc68db32 = async (migrationName)=>{
    try {
        await $a272df753f76392f$export$b9c77372957d1829();
        await $ba05d75a0e232cea$export$e772c8ff12451969(2000);
        await $a272df753f76392f$export$1fe1303d5ebdc741(migrationName);
    } catch (error) {
        $e7da8e8b58079630$export$af88d00dbe7f521.error(`Failed to run migrations: ${error.command}`);
        $e7da8e8b58079630$export$af88d00dbe7f521.newLine();
        $e7da8e8b58079630$export$af88d00dbe7f521.warning("You can manually run migrations with the following commands:");
        $e7da8e8b58079630$export$af88d00dbe7f521.list([
            `npm run docker:start - ${$e7da8e8b58079630$export$af88d00dbe7f521.withMeta("starts a local docker instance with your database")}`,
            `npm run prisma:migrate:dev - ${$e7da8e8b58079630$export$af88d00dbe7f521.withMeta("runs db migrations")}`, 
        ]);
        $e7da8e8b58079630$export$af88d00dbe7f521.newLine();
    }
};


var $ecb5f2cca8130514$exports = {};

$parcel$export($ecb5f2cca8130514$exports, "prompts", () => $ecb5f2cca8130514$export$49bc81080f205727);

var $20b1a31b93e300d6$exports = {};

$parcel$export($20b1a31b93e300d6$exports, "throwHandledError", () => $20b1a31b93e300d6$export$bbec2b7d78f06a8d);
$parcel$export($20b1a31b93e300d6$exports, "onPromptCancel", () => $20b1a31b93e300d6$export$255442a2e70017bf);
const $20b1a31b93e300d6$export$bbec2b7d78f06a8d = ({ command: command , message: message , error: error  })=>{
    if (error) console.error(error);
    throw {
        command: command,
        message: message
    };
};
const $20b1a31b93e300d6$export$255442a2e70017bf = ()=>{
    $20b1a31b93e300d6$export$bbec2b7d78f06a8d({
        command: "prompt",
        message: "User cancelled step"
    });
};


const $ecb5f2cca8130514$export$49bc81080f205727 = (questions, options)=>{
    return ($parcel$interopDefault($12jGn$prompts))(questions, {
        ...options,
        onCancel: ()=>{
            $20b1a31b93e300d6$export$bbec2b7d78f06a8d({
                command: "prompt",
                message: "Cancelled by user"
            });
        }
    });
};


var $c23c90dbb6f61999$exports = {};

$parcel$export($c23c90dbb6f61999$exports, "GeneratorLocation", () => $c23c90dbb6f61999$export$84148c2dee5b43c5);
var $c23c90dbb6f61999$export$84148c2dee5b43c5;
(function(GeneratorLocation) {
    GeneratorLocation[GeneratorLocation["Local"] = 0] = "Local";
    GeneratorLocation[GeneratorLocation["Remote"] = 1] = "Remote";
})($c23c90dbb6f61999$export$84148c2dee5b43c5 || ($c23c90dbb6f61999$export$84148c2dee5b43c5 = {
}));


var $40eb5dc350e81bd1$exports = {};

$parcel$export($40eb5dc350e81bd1$exports, "filetypeToPrettierParser", () => $40eb5dc350e81bd1$export$d31b835ca3ad9ed);
$parcel$export($40eb5dc350e81bd1$exports, "getPrettierParser", () => $40eb5dc350e81bd1$export$b0ee1b8c2e95dcbe);
$parcel$export($40eb5dc350e81bd1$exports, "transformPrettier", () => $40eb5dc350e81bd1$export$fe3204bf5faa7e51);

const $40eb5dc350e81bd1$export$d31b835ca3ad9ed = {
    ts: "typescript",
    tsx: "typescript",
    js: "babel",
    jsx: "babel",
    json: "json",
    yml: "yaml",
    yaml: "yaml",
    md: "markdown",
    html: "html",
    css: "css",
    scss: "scss",
    sass: "sass",
    less: "less",
    graphql: "graphql"
};
const $40eb5dc350e81bd1$export$b0ee1b8c2e95dcbe = (filePath)=>{
    const filetype = filePath.slice(filePath.lastIndexOf(".") + 1);
    return $40eb5dc350e81bd1$export$d31b835ca3ad9ed[filetype] || null;
};
const $40eb5dc350e81bd1$export$fe3204bf5faa7e51 = (filePath)=>async (source)=>{
        const parser = $40eb5dc350e81bd1$export$b0ee1b8c2e95dcbe(filePath);
        if (parser) return ($parcel$interopDefault($12jGn$prettier)).format(source, {
            parser: parser
        });
        return source;
    }
;


var $fbe59e8b6e59844e$exports = {};

$parcel$export($fbe59e8b6e59844e$exports, "transformEjs", () => $fbe59e8b6e59844e$export$4a5dd64dec194a6a);

const $fbe59e8b6e59844e$export$4a5dd64dec194a6a = async (source, context)=>{
    return ($parcel$interopDefault($12jGn$ejs)).render(source, context);
};


var $76d21e206339830d$exports = {};

$parcel$export($76d21e206339830d$exports, "createJscodeshiftProgram", () => $76d21e206339830d$export$259b2dad08caae9f);
$parcel$export($76d21e206339830d$exports, "createJscodeshiftTransform", () => $76d21e206339830d$export$9b3cf6a05365ed8f);

const $76d21e206339830d$export$259b2dad08caae9f = (source)=>{
    const j = ($parcel$interopDefault($12jGn$jscodeshift)).withParser("tsx");
    return {
        program: j(source),
        jscodeshift: ($parcel$interopDefault($12jGn$jscodeshift)),
        j: j
    };
};
const $76d21e206339830d$export$9b3cf6a05365ed8f = (transform)=>async (source, context)=>{
        let { program: program  } = $76d21e206339830d$export$259b2dad08caae9f(source);
        program = await transform(program, context);
        return program.toSource();
    }
;


var $7c7bcae8b4b4a474$exports = {};

$parcel$export($7c7bcae8b4b4a474$exports, "addImport", () => $7c7bcae8b4b4a474$export$d41ce78d713db213);

const $7c7bcae8b4b4a474$export$d41ce78d713db213 = (program, importToAdd)=>{
    const importStatementCount = program.find(($parcel$interopDefault($12jGn$jscodeshift)).ImportDeclaration).length;
    if (importStatementCount === 0) {
        program.find(($parcel$interopDefault($12jGn$jscodeshift)).Statement).at(0).insertBefore(importToAdd);
        return program;
    }
    try {
        program.find(($parcel$interopDefault($12jGn$jscodeshift)).ImportDeclaration).forEach((stmt, idx)=>{
            const importValue = stmt.value.source.value;
            if (importToAdd.source.value === importValue) {
                console.error(`Skipping addImport, duplicate import found: ${importValue}`);
                throw new Error();
            } else if (idx === importStatementCount - 1) stmt.replace(stmt.node, importToAdd);
        });
    } catch (e) {
    }
    return program;
};


var $a785308a1f33fdee$exports = {};

$parcel$export($a785308a1f33fdee$exports, "findModuleExportsExpressions", () => $a785308a1f33fdee$export$911865420682240f);

const $a785308a1f33fdee$export$911865420682240f = (program)=>program.find(($parcel$interopDefault($12jGn$jscodeshift)).AssignmentExpression).filter((path)=>{
        const { left: left , right: right  } = path.value;
        return left.type === "MemberExpression" && left.object.type === "Identifier" && left.property.type === "Identifier" && left.property.name === "exports" && right.type === "ObjectExpression";
    })
;


var $3988e7c8ec377a33$exports = {};

$parcel$export($3988e7c8ec377a33$exports, "addPrismaEnum", () => $3988e7c8ec377a33$export$cca4c4c5e6e9f10);
var $d69e7b2b0b4e03f5$exports = {};

$parcel$export($d69e7b2b0b4e03f5$exports, "produceSchema", () => $d69e7b2b0b4e03f5$export$b0848f155f13a730);

const $d69e7b2b0b4e03f5$export$b0848f155f13a730 = async (source, producer)=>{
    const schema = await $12jGn$mrleeboprismaast.getSchema(source);
    producer(schema);
    return $12jGn$mrleeboprismaast.printSchema(schema);
};


const $3988e7c8ec377a33$export$cca4c4c5e6e9f10 = (source, enumProps)=>$d69e7b2b0b4e03f5$export$b0848f155f13a730(source, (schema)=>{
        const existing = schema.list.find((x)=>x.type === "enum" && x.name === enumProps.name
        );
        existing ? Object.assign(existing, enumProps) : schema.list.push(enumProps);
    })
;


var $160f696d4998c975$exports = {};

$parcel$export($160f696d4998c975$exports, "addPrismaField", () => $160f696d4998c975$export$8eb5c4001bc26e32);

const $160f696d4998c975$export$8eb5c4001bc26e32 = (source, modelName, fieldProps)=>$d69e7b2b0b4e03f5$export$b0848f155f13a730(source, (schema)=>{
        const model = schema.list.find((x)=>x.type === "model" && x.name === modelName
        );
        if (!model) return;
        const existing = model.properties.find((x)=>x.type === "field" && x.name === fieldProps.name
        );
        existing ? Object.assign(existing, fieldProps) : model.properties.push(fieldProps);
    })
;


var $01ac6eec854e7b86$exports = {};

$parcel$export($01ac6eec854e7b86$exports, "addPrismaGenerator", () => $01ac6eec854e7b86$export$3af34c32d299916);

const $01ac6eec854e7b86$export$3af34c32d299916 = (source, generatorProps)=>$d69e7b2b0b4e03f5$export$b0848f155f13a730(source, (schema)=>{
        const existing = schema.list.find((x)=>x.type === "generator" && x.name === generatorProps.name
        );
        existing ? Object.assign(existing, generatorProps) : schema.list.push(generatorProps);
    })
;


var $a0a0a02d522fff0d$exports = {};

$parcel$export($a0a0a02d522fff0d$exports, "addPrismaModelAttribute", () => $a0a0a02d522fff0d$export$a5156224d356ba0e);

const $a0a0a02d522fff0d$export$a5156224d356ba0e = (source, modelName, attributeProps)=>$d69e7b2b0b4e03f5$export$b0848f155f13a730(source, (schema)=>{
        const model = schema.list.find((x)=>x.type === "model" && x.name === modelName
        );
        if (!model) return;
        const existing = model.properties.find((x)=>x.type === "attribute" && x.name === attributeProps.name
        );
        existing ? Object.assign(existing, attributeProps) : model.properties.push(attributeProps);
    })
;


var $69a39e76d57f30ae$exports = {};

$parcel$export($69a39e76d57f30ae$exports, "addPrismaModel", () => $69a39e76d57f30ae$export$c9297bb316159832);

const $69a39e76d57f30ae$export$c9297bb316159832 = (source, modelProps)=>$d69e7b2b0b4e03f5$export$b0848f155f13a730(source, (schema)=>{
        const existing = schema.list.find((x)=>x.type === "model" && x.name === modelProps.name
        );
        existing ? Object.assign(existing, modelProps) : schema.list.push(modelProps);
    })
;



var $00de315d5fb15e86$exports = {};

$parcel$export($00de315d5fb15e86$exports, "setPrismaDataSource", () => $00de315d5fb15e86$export$12acaf1194b381ec);

const $00de315d5fb15e86$export$12acaf1194b381ec = (source, datasourceProps)=>$d69e7b2b0b4e03f5$export$b0848f155f13a730(source, (schema)=>{
        const existing = schema.list.find((x)=>x.type === "datasource"
        );
        existing ? Object.assign(existing, datasourceProps) : schema.list.push(datasourceProps);
    })
;


var $7ae182e6c0803992$exports = {};

$parcel$export($7ae182e6c0803992$exports, "logSchema", () => $7ae182e6c0803992$export$de6b7d32e7276555);


const $7ae182e6c0803992$export$de6b7d32e7276555 = async (source)=>{
    const schema = await $12jGn$mrleeboprismaast.getSchema(source);
    const printPath = `${process.cwd()}/schema.json`;
    console.log(schema);
    console.log(`written to ${printPath}`);
    ($parcel$interopDefault($12jGn$fsextra)).writeFileSync(printPath, JSON.stringify(schema, null, 2));
};




var $2d18eede51a6b325$exports = {};

$parcel$export($2d18eede51a6b325$exports, "isWriteable", () => $2d18eede51a6b325$export$2f93ca0bbcabd05f);
$parcel$export($2d18eede51a6b325$exports, "makeDir", () => $2d18eede51a6b325$export$ef2527cb59c03af5);
$parcel$export($2d18eede51a6b325$exports, "removeDir", () => $2d18eede51a6b325$export$75426ec8836e4c8);
$parcel$export($2d18eede51a6b325$exports, "getFilePaths", () => $2d18eede51a6b325$export$9f499d9b98b35925);
$parcel$export($2d18eede51a6b325$exports, "copyFiles", () => $2d18eede51a6b325$export$6738b3634be42910);
$parcel$export($2d18eede51a6b325$exports, "fileExists", () => $2d18eede51a6b325$export$ffc21166d570a16);
$parcel$export($2d18eede51a6b325$exports, "readJson", () => $2d18eede51a6b325$export$6a368473377ea45c);
$parcel$export($2d18eede51a6b325$exports, "writeJson", () => $2d18eede51a6b325$export$476f628a921f7411);
$parcel$export($2d18eede51a6b325$exports, "readFile", () => $2d18eede51a6b325$export$72c04af63de9061a);
$parcel$export($2d18eede51a6b325$exports, "writeFile", () => $2d18eede51a6b325$export$552bfb764b5cd2b4);
$parcel$export($2d18eede51a6b325$exports, "runTransforms", () => $2d18eede51a6b325$export$c88707f9397b495c);



const $2d18eede51a6b325$export$2f93ca0bbcabd05f = async (directory)=>{
    try {
        await ($parcel$interopDefault($12jGn$fsextra)).promises.access(directory, (($parcel$interopDefault($12jGn$fsextra)).constants || ($parcel$interopDefault($12jGn$fsextra))).W_OK);
        return true;
    } catch (err) {
        return false;
    }
};
const $2d18eede51a6b325$export$ef2527cb59c03af5 = (root, options = {
    recursive: true
})=>($parcel$interopDefault($12jGn$fsextra)).mkdirSync(root, options)
;
const $2d18eede51a6b325$export$75426ec8836e4c8 = (path)=>($parcel$interopDefault($12jGn$rimraf)).sync(path)
;
const $2d18eede51a6b325$export$9f499d9b98b35925 = (patterns, options)=>($parcel$interopDefault($12jGn$globby))(patterns, options)
;
const $2d18eede51a6b325$export$6738b3634be42910 = (source, destination)=>($parcel$interopDefault($12jGn$fsextra)).copy(source, destination)
;
const $2d18eede51a6b325$export$ffc21166d570a16 = (path)=>($parcel$interopDefault($12jGn$fsextra)).existsSync(path)
;
const $2d18eede51a6b325$export$6a368473377ea45c = (...args)=>($parcel$interopDefault($12jGn$fsextra)).readJsonSync(...args)
;
const $2d18eede51a6b325$export$476f628a921f7411 = (...args)=>($parcel$interopDefault($12jGn$fsextra)).outputJSONSync(...args)
;
const $2d18eede51a6b325$export$72c04af63de9061a = (filePath)=>($parcel$interopDefault($12jGn$fsextra)).readFileSync(filePath, "utf8")
;
const $2d18eede51a6b325$export$552bfb764b5cd2b4 = (filePath, content, options = {
    force: false,
    append: false
})=>{
    const exists = ($parcel$interopDefault($12jGn$fsextra)).existsSync(filePath);
    if (exists && options.append) return ($parcel$interopDefault($12jGn$fsextra)).appendFileSync(filePath, content);
    if (!exists || exists && options.force) return ($parcel$interopDefault($12jGn$fsextra)).outputFileSync(filePath, content);
    throw new Error(`Cannot write file to ${filePath}`);
};
const $2d18eede51a6b325$export$c88707f9397b495c = async (filePath, ...transforms)=>{
    const isStringPath = typeof filePath === "string";
    const sourcePath = isStringPath ? filePath : filePath.src;
    const destinationPath = isStringPath ? filePath : filePath.dest;
    let source = ($parcel$interopDefault($12jGn$fsextra)).readFileSync(sourcePath, "utf8");
    for (const [transform, context] of transforms){
        const transformed = await transform(source, context);
        if (transformed) source = transformed;
    }
    if (destinationPath) return ($parcel$interopDefault($12jGn$fsextra)).outputFile(destinationPath, source);
    return source;
};



var $f9bdbe84adff927d$exports = {};

$parcel$export($f9bdbe84adff927d$exports, "checkAndNotifyUpdates", () => $f9bdbe84adff927d$export$85481a6fc31b788c);


const $f9bdbe84adff927d$export$85481a6fc31b788c = async (packageJson)=>{
    try {
        const res = await ($parcel$interopDefault($12jGn$updatecheck))(packageJson).catch(()=>null
        );
        if (res === null || res === void 0 ? void 0 : res.latest) {
            $e7da8e8b58079630$export$af88d00dbe7f521.newLine();
            $e7da8e8b58079630$export$af88d00dbe7f521.warning(`A new version of "${packageJson.name}" is available!`);
            $e7da8e8b58079630$export$af88d00dbe7f521.log(`You can update by running: ${$e7da8e8b58079630$export$af88d00dbe7f521.withVariable(`npm update ${packageJson.name}`)}`);
            $e7da8e8b58079630$export$af88d00dbe7f521.newLine();
        }
        process.exit();
    } catch  {
    // ignore error
    }
};


$parcel$exportWildcard(module.exports, $40b2ad338fe99ac0$exports);
$parcel$exportWildcard(module.exports, $a272df753f76392f$exports);
$parcel$exportWildcard(module.exports, $ecb5f2cca8130514$exports);
$parcel$exportWildcard(module.exports, $c23c90dbb6f61999$exports);
$parcel$exportWildcard(module.exports, $40eb5dc350e81bd1$exports);
$parcel$exportWildcard(module.exports, $fbe59e8b6e59844e$exports);
$parcel$exportWildcard(module.exports, $76d21e206339830d$exports);
$parcel$exportWildcard(module.exports, $7c7bcae8b4b4a474$exports);
$parcel$exportWildcard(module.exports, $a785308a1f33fdee$exports);
$parcel$exportWildcard(module.exports, $3988e7c8ec377a33$exports);
$parcel$exportWildcard(module.exports, $160f696d4998c975$exports);
$parcel$exportWildcard(module.exports, $01ac6eec854e7b86$exports);
$parcel$exportWildcard(module.exports, $a0a0a02d522fff0d$exports);
$parcel$exportWildcard(module.exports, $69a39e76d57f30ae$exports);
$parcel$exportWildcard(module.exports, $d69e7b2b0b4e03f5$exports);
$parcel$exportWildcard(module.exports, $00de315d5fb15e86$exports);
$parcel$exportWildcard(module.exports, $7ae182e6c0803992$exports);
$parcel$exportWildcard(module.exports, $20b1a31b93e300d6$exports);
$parcel$exportWildcard(module.exports, $ba05d75a0e232cea$exports);
$parcel$exportWildcard(module.exports, $2d18eede51a6b325$exports);
$parcel$exportWildcard(module.exports, $e7da8e8b58079630$exports);
$parcel$exportWildcard(module.exports, $f9bdbe84adff927d$exports);
