// @founding/devkit/dist/index.mjs
import $7gN5q$crossspawn from "cross-spawn";
import $7gN5q$chalk from "chalk";
import {Table as $7gN5q$Table} from "console-table-printer";
import $7gN5q$ora from "ora";
import $7gN5q$readline from "readline";
import $7gN5q$prompts from "prompts";
import $7gN5q$prettier from "prettier";
import $7gN5q$ejs from "ejs";
import $7gN5q$jscodeshift from "jscodeshift";
import {getSchema as $7gN5q$getSchema, printSchema as $7gN5q$printSchema} from "@mrleebo/prisma-ast";
import $7gN5q$fsextra from "fs-extra";
import $7gN5q$globby from "globby";
import $7gN5q$rimraf from "rimraf";
import $7gN5q$updatecheck from "update-check";

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
var $d2784300cb69549f$exports = {};

$parcel$export($d2784300cb69549f$exports, "filterDependencies", () => $d2784300cb69549f$export$a48cded481a7b9a3);
$parcel$export($d2784300cb69549f$exports, "filterTemplates", () => $d2784300cb69549f$export$883cc5315f212021);
const $d2784300cb69549f$var$filterList = (list, context)=>{
    return list.filter(({ filters: filters  })=>{
        if (filters) return filters.every((filter)=>filter(context)
        );
        return true;
    });
};
const $d2784300cb69549f$export$a48cded481a7b9a3 = (dependencies, context)=>{
    return $d2784300cb69549f$var$filterList(dependencies, context);
};
const $d2784300cb69549f$export$883cc5315f212021 = (templates, context)=>{
    return $d2784300cb69549f$var$filterList(templates, context);
};


var $f26fe0275fa4931b$exports = {};

$parcel$export($f26fe0275fa4931b$exports, "startDocker", () => $f26fe0275fa4931b$export$b9c77372957d1829);
$parcel$export($f26fe0275fa4931b$exports, "stopDocker", () => $f26fe0275fa4931b$export$9238c7a74ca8f573);
$parcel$export($f26fe0275fa4931b$exports, "runMigrations", () => $f26fe0275fa4931b$export$1fe1303d5ebdc741);
$parcel$export($f26fe0275fa4931b$exports, "runPrismaCodegen", () => $f26fe0275fa4931b$export$85f637821bb9f000);
$parcel$export($f26fe0275fa4931b$exports, "syncGeneratorMigrations", () => $f26fe0275fa4931b$export$d1a4d70edc68db32);
var $e68a37bf8acc13df$exports = {};

$parcel$export($e68a37bf8acc13df$exports, "execSync", () => $e68a37bf8acc13df$export$fa77af024d7e00a8);
$parcel$export($e68a37bf8acc13df$exports, "exec", () => $e68a37bf8acc13df$export$78e3044358792147);
$parcel$export($e68a37bf8acc13df$exports, "sleep", () => $e68a37bf8acc13df$export$e772c8ff12451969);

const $e68a37bf8acc13df$export$fa77af024d7e00a8 = (command, args, options = {
})=>{
    return $7gN5q$crossspawn.sync(command, args, options);
};
const $e68a37bf8acc13df$export$78e3044358792147 = (command, args, withStdio = true)=>new Promise((resolve, reject)=>{
        const child = $7gN5q$crossspawn(command, args, {
            stdio: withStdio ? "inherit" : "ignore",
            env: {
                ...process.env,
                ADBLOCK: "1",
                DISABLE_OPENCOLLECTIVE: "1",
                TS_NODE_COMPILER_OPTIONS: JSON.stringify({
                    module: "commonjs"
                })
            }
        });
        child.on("close", (code)=>{
            if (code !== 0) {
                reject({
                    command: `${command} ${args.join(" ")}`
                });
                return;
            }
            resolve();
        });
    })
;
const $e68a37bf8acc13df$export$e772c8ff12451969 = (ms)=>{
    return new Promise((resolve)=>setTimeout(resolve, ms)
    );
};


var $62ae51800c28537f$exports = {};

$parcel$export($62ae51800c28537f$exports, "table", () => $62ae51800c28537f$export$9852986a3ec5f6a0);
$parcel$export($62ae51800c28537f$exports, "chalk", () => $62ae51800c28537f$export$a0b6f61a94abd245);
$parcel$export($62ae51800c28537f$exports, "logger", () => $62ae51800c28537f$export$af88d00dbe7f521);




const $62ae51800c28537f$export$9852986a3ec5f6a0 = $7gN5q$Table;
const $62ae51800c28537f$export$a0b6f61a94abd245 = $7gN5q$chalk;
const $62ae51800c28537f$export$af88d00dbe7f521 = {
    withWarning (str) {
        return `⚠️  ${$7gN5q$chalk.yellow(str)}`;
    },
    withCaret (str) {
        return `${$7gN5q$chalk.gray(">")} ${str}`;
    },
    withCheck (str) {
        return `${$7gN5q$chalk.green("✔")} ${str}`;
    },
    withX (str) {
        return `${$7gN5q$chalk.red.bold("✕")} ${str}`;
    },
    withProgress (str) {
        return $62ae51800c28537f$export$af88d00dbe7f521.withCaret(str);
    },
    withError (str) {
        return $62ae51800c28537f$export$af88d00dbe7f521.withX($7gN5q$chalk.red.bold(str));
    },
    withMeta (str) {
        return $7gN5q$chalk.dim(str);
    },
    withVariable (str) {
        return $7gN5q$chalk.green(`${str}`);
    },
    withCommand (str) {
        return $7gN5q$chalk.cyan(`${str}`);
    },
    withIndent (str, indent = 0) {
        const indents = "   ".repeat(indent);
        return `${indents}${str}`;
    },
    spinner (str) {
        const spinner = $7gN5q$ora({
            text: str,
            color: "cyan",
            spinner: "dots"
        });
        if (str) spinner.start();
        return spinner;
    },
    clearLine (msg) {
        $7gN5q$readline.clearLine(process.stdout, 0);
        $7gN5q$readline.cursorTo(process.stdout, 0);
        msg && process.stdout.write(msg);
    },
    clearConsole () {
        if (process.platform === "win32") process.stdout.write("\x1B[2J\x1B[0f");
        else process.stdout.write("\x1B[2J\x1B[3J\x1B[H");
    },
    title (msg) {
        console.log($7gN5q$chalk.bold(msg));
    },
    list (msgs) {
        console.log(msgs.map((m)=>$62ae51800c28537f$export$af88d00dbe7f521.withCaret(m)
        ).join("\n"));
    },
    error (msg) {
        console.error($62ae51800c28537f$export$af88d00dbe7f521.withX($7gN5q$chalk.red.bold(msg)));
    },
    warning (msg) {
        console.log($62ae51800c28537f$export$af88d00dbe7f521.withWarning(msg));
    },
    meta (msg) {
        console.log($62ae51800c28537f$export$af88d00dbe7f521.withMeta(msg));
    },
    progress (msg) {
        console.log($62ae51800c28537f$export$af88d00dbe7f521.withCaret(msg));
    },
    success (msg) {
        console.log($62ae51800c28537f$export$af88d00dbe7f521.withCheck(msg));
    },
    log (msg, indents = 0, newLine = false) {
        console.log($62ae51800c28537f$export$af88d00dbe7f521.withIndent(msg, indents));
        if (newLine) console.log();
    },
    newLine () {
        console.log();
    },
    table: $62ae51800c28537f$export$9852986a3ec5f6a0
};


const $f26fe0275fa4931b$export$b9c77372957d1829 = async ()=>{
    console.log("Starting docker instance");
    await $e68a37bf8acc13df$export$78e3044358792147("docker-compose", [
        "up",
        "-d"
    ]);
};
const $f26fe0275fa4931b$export$9238c7a74ca8f573 = async ()=>{
    console.log("Stopping docker instance");
    await $e68a37bf8acc13df$export$78e3044358792147("docker-compose", [
        "down"
    ]);
};
const $f26fe0275fa4931b$export$1fe1303d5ebdc741 = async (name = "")=>{
    console.log("Running local migrations");
    await $e68a37bf8acc13df$export$78e3044358792147("npm", [
        "run",
        "prisma:migrate:dev",
        "--",
        "-n",
        name,
        "--skip-seed", 
    ]);
};
const $f26fe0275fa4931b$export$85f637821bb9f000 = async ()=>{
    console.log("Running prisma codegen");
    await $e68a37bf8acc13df$export$78e3044358792147("npm", [
        "run",
        "prisma:generate"
    ]);
};
const $f26fe0275fa4931b$export$d1a4d70edc68db32 = async (migrationName)=>{
    try {
        await $f26fe0275fa4931b$export$b9c77372957d1829();
        await $e68a37bf8acc13df$export$e772c8ff12451969(2000);
        await $f26fe0275fa4931b$export$1fe1303d5ebdc741(migrationName);
    } catch (error) {
        $62ae51800c28537f$export$af88d00dbe7f521.error(`Failed to run migrations: ${error.command}`);
        $62ae51800c28537f$export$af88d00dbe7f521.newLine();
        $62ae51800c28537f$export$af88d00dbe7f521.warning("You can manually run migrations with the following commands:");
        $62ae51800c28537f$export$af88d00dbe7f521.list([
            `npm run docker:start - ${$62ae51800c28537f$export$af88d00dbe7f521.withMeta("starts a local docker instance with your database")}`,
            `npm run prisma:migrate:dev - ${$62ae51800c28537f$export$af88d00dbe7f521.withMeta("runs db migrations")}`, 
        ]);
        $62ae51800c28537f$export$af88d00dbe7f521.newLine();
    }
};


var $e35cfb01f3b02780$exports = {};

$parcel$export($e35cfb01f3b02780$exports, "prompts", () => $e35cfb01f3b02780$export$49bc81080f205727);

var $1bdf442fb86e9adf$exports = {};

$parcel$export($1bdf442fb86e9adf$exports, "throwHandledError", () => $1bdf442fb86e9adf$export$bbec2b7d78f06a8d);
$parcel$export($1bdf442fb86e9adf$exports, "onPromptCancel", () => $1bdf442fb86e9adf$export$255442a2e70017bf);
const $1bdf442fb86e9adf$export$bbec2b7d78f06a8d = ({ command: command , message: message , error: error  })=>{
    if (error) console.error(error);
    throw {
        command: command,
        message: message
    };
};
const $1bdf442fb86e9adf$export$255442a2e70017bf = ()=>{
    $1bdf442fb86e9adf$export$bbec2b7d78f06a8d({
        command: "prompt",
        message: "User cancelled step"
    });
};


const $e35cfb01f3b02780$export$49bc81080f205727 = (questions, options)=>{
    return $7gN5q$prompts(questions, {
        ...options,
        onCancel: ()=>{
            $1bdf442fb86e9adf$export$bbec2b7d78f06a8d({
                command: "prompt",
                message: "Cancelled by user"
            });
        }
    });
};


var $910700bd48d72431$exports = {};

$parcel$export($910700bd48d72431$exports, "GeneratorLocation", () => $910700bd48d72431$export$84148c2dee5b43c5);
var $910700bd48d72431$export$84148c2dee5b43c5;
(function(GeneratorLocation) {
    GeneratorLocation[GeneratorLocation["Local"] = 0] = "Local";
    GeneratorLocation[GeneratorLocation["Remote"] = 1] = "Remote";
})($910700bd48d72431$export$84148c2dee5b43c5 || ($910700bd48d72431$export$84148c2dee5b43c5 = {
}));


var $83f9063289dc0194$exports = {};

$parcel$export($83f9063289dc0194$exports, "filetypeToPrettierParser", () => $83f9063289dc0194$export$d31b835ca3ad9ed);
$parcel$export($83f9063289dc0194$exports, "getPrettierParser", () => $83f9063289dc0194$export$b0ee1b8c2e95dcbe);
$parcel$export($83f9063289dc0194$exports, "transformPrettier", () => $83f9063289dc0194$export$fe3204bf5faa7e51);

const $83f9063289dc0194$export$d31b835ca3ad9ed = {
    ts: "typescript",
    tsx: "typescript",
    js: "babel",
    jsx: "babel",
    json: "json",
    yml: "yaml",
    yaml: "yaml",
    md: "markdown",
    html: "html",
    css: "css",
    scss: "scss",
    sass: "sass",
    less: "less",
    graphql: "graphql"
};
const $83f9063289dc0194$export$b0ee1b8c2e95dcbe = (filePath)=>{
    const filetype = filePath.slice(filePath.lastIndexOf(".") + 1);
    return $83f9063289dc0194$export$d31b835ca3ad9ed[filetype] || null;
};
const $83f9063289dc0194$export$fe3204bf5faa7e51 = (filePath)=>async (source)=>{
        const parser = $83f9063289dc0194$export$b0ee1b8c2e95dcbe(filePath);
        if (parser) return $7gN5q$prettier.format(source, {
            parser: parser
        });
        return source;
    }
;


var $de00a81b1a4c70e1$exports = {};

$parcel$export($de00a81b1a4c70e1$exports, "transformEjs", () => $de00a81b1a4c70e1$export$4a5dd64dec194a6a);

const $de00a81b1a4c70e1$export$4a5dd64dec194a6a = async (source, context)=>{
    return $7gN5q$ejs.render(source, context);
};


var $f3a3e1ac784d319c$exports = {};

$parcel$export($f3a3e1ac784d319c$exports, "createJscodeshiftProgram", () => $f3a3e1ac784d319c$export$259b2dad08caae9f);
$parcel$export($f3a3e1ac784d319c$exports, "createJscodeshiftTransform", () => $f3a3e1ac784d319c$export$9b3cf6a05365ed8f);

const $f3a3e1ac784d319c$export$259b2dad08caae9f = (source)=>{
    const j = $7gN5q$jscodeshift.withParser("tsx");
    return {
        program: j(source),
        jscodeshift: $7gN5q$jscodeshift,
        j: j
    };
};
const $f3a3e1ac784d319c$export$9b3cf6a05365ed8f = (transform)=>async (source, context)=>{
        let { program: program  } = $f3a3e1ac784d319c$export$259b2dad08caae9f(source);
        program = await transform(program, context);
        return program.toSource();
    }
;


var $17d5a96dac1073ea$exports = {};

$parcel$export($17d5a96dac1073ea$exports, "addImport", () => $17d5a96dac1073ea$export$d41ce78d713db213);

const $17d5a96dac1073ea$export$d41ce78d713db213 = (program, importToAdd)=>{
    const importStatementCount = program.find($7gN5q$jscodeshift.ImportDeclaration).length;
    if (importStatementCount === 0) {
        program.find($7gN5q$jscodeshift.Statement).at(0).insertBefore(importToAdd);
        return program;
    }
    try {
        program.find($7gN5q$jscodeshift.ImportDeclaration).forEach((stmt, idx)=>{
            const importValue = stmt.value.source.value;
            if (importToAdd.source.value === importValue) {
                console.error(`Skipping addImport, duplicate import found: ${importValue}`);
                throw new Error();
            } else if (idx === importStatementCount - 1) stmt.replace(stmt.node, importToAdd);
        });
    } catch (e) {
    }
    return program;
};


var $8e7ad5eba6db344f$exports = {};

$parcel$export($8e7ad5eba6db344f$exports, "findModuleExportsExpressions", () => $8e7ad5eba6db344f$export$911865420682240f);

const $8e7ad5eba6db344f$export$911865420682240f = (program)=>program.find($7gN5q$jscodeshift.AssignmentExpression).filter((path)=>{
        const { left: left , right: right  } = path.value;
        return left.type === "MemberExpression" && left.object.type === "Identifier" && left.property.type === "Identifier" && left.property.name === "exports" && right.type === "ObjectExpression";
    })
;


var $8235f45a91cc1e6c$exports = {};

$parcel$export($8235f45a91cc1e6c$exports, "addPrismaEnum", () => $8235f45a91cc1e6c$export$cca4c4c5e6e9f10);
var $6fd0cc407948fe8e$exports = {};

$parcel$export($6fd0cc407948fe8e$exports, "produceSchema", () => $6fd0cc407948fe8e$export$b0848f155f13a730);

const $6fd0cc407948fe8e$export$b0848f155f13a730 = async (source, producer)=>{
    const schema = await $7gN5q$getSchema(source);
    producer(schema);
    return $7gN5q$printSchema(schema);
};


const $8235f45a91cc1e6c$export$cca4c4c5e6e9f10 = (source, enumProps)=>$6fd0cc407948fe8e$export$b0848f155f13a730(source, (schema)=>{
        const existing = schema.list.find((x)=>x.type === "enum" && x.name === enumProps.name
        );
        existing ? Object.assign(existing, enumProps) : schema.list.push(enumProps);
    })
;


var $eedf22dd7a0eade6$exports = {};

$parcel$export($eedf22dd7a0eade6$exports, "addPrismaField", () => $eedf22dd7a0eade6$export$8eb5c4001bc26e32);

const $eedf22dd7a0eade6$export$8eb5c4001bc26e32 = (source, modelName, fieldProps)=>$6fd0cc407948fe8e$export$b0848f155f13a730(source, (schema)=>{
        const model = schema.list.find((x)=>x.type === "model" && x.name === modelName
        );
        if (!model) return;
        const existing = model.properties.find((x)=>x.type === "field" && x.name === fieldProps.name
        );
        existing ? Object.assign(existing, fieldProps) : model.properties.push(fieldProps);
    })
;


var $919f5877cffaeae0$exports = {};

$parcel$export($919f5877cffaeae0$exports, "addPrismaGenerator", () => $919f5877cffaeae0$export$3af34c32d299916);

const $919f5877cffaeae0$export$3af34c32d299916 = (source, generatorProps)=>$6fd0cc407948fe8e$export$b0848f155f13a730(source, (schema)=>{
        const existing = schema.list.find((x)=>x.type === "generator" && x.name === generatorProps.name
        );
        existing ? Object.assign(existing, generatorProps) : schema.list.push(generatorProps);
    })
;


var $87517b8de60d215a$exports = {};

$parcel$export($87517b8de60d215a$exports, "addPrismaModelAttribute", () => $87517b8de60d215a$export$a5156224d356ba0e);

const $87517b8de60d215a$export$a5156224d356ba0e = (source, modelName, attributeProps)=>$6fd0cc407948fe8e$export$b0848f155f13a730(source, (schema)=>{
        const model = schema.list.find((x)=>x.type === "model" && x.name === modelName
        );
        if (!model) return;
        const existing = model.properties.find((x)=>x.type === "attribute" && x.name === attributeProps.name
        );
        existing ? Object.assign(existing, attributeProps) : model.properties.push(attributeProps);
    })
;


var $8b3c464caa8bcb73$exports = {};

$parcel$export($8b3c464caa8bcb73$exports, "addPrismaModel", () => $8b3c464caa8bcb73$export$c9297bb316159832);

const $8b3c464caa8bcb73$export$c9297bb316159832 = (source, modelProps)=>$6fd0cc407948fe8e$export$b0848f155f13a730(source, (schema)=>{
        const existing = schema.list.find((x)=>x.type === "model" && x.name === modelProps.name
        );
        existing ? Object.assign(existing, modelProps) : schema.list.push(modelProps);
    })
;



var $be99dba2948f5913$exports = {};

$parcel$export($be99dba2948f5913$exports, "setPrismaDataSource", () => $be99dba2948f5913$export$12acaf1194b381ec);

const $be99dba2948f5913$export$12acaf1194b381ec = (source, datasourceProps)=>$6fd0cc407948fe8e$export$b0848f155f13a730(source, (schema)=>{
        const existing = schema.list.find((x)=>x.type === "datasource"
        );
        existing ? Object.assign(existing, datasourceProps) : schema.list.push(datasourceProps);
    })
;


var $055006d47df8621a$exports = {};

$parcel$export($055006d47df8621a$exports, "logSchema", () => $055006d47df8621a$export$de6b7d32e7276555);


const $055006d47df8621a$export$de6b7d32e7276555 = async (source)=>{
    const schema = await $7gN5q$getSchema(source);
    const printPath = `${process.cwd()}/schema.json`;
    console.log(schema);
    console.log(`written to ${printPath}`);
    $7gN5q$fsextra.writeFileSync(printPath, JSON.stringify(schema, null, 2));
};




var $c8f7705cf7ea7419$exports = {};

$parcel$export($c8f7705cf7ea7419$exports, "isWriteable", () => $c8f7705cf7ea7419$export$2f93ca0bbcabd05f);
$parcel$export($c8f7705cf7ea7419$exports, "makeDir", () => $c8f7705cf7ea7419$export$ef2527cb59c03af5);
$parcel$export($c8f7705cf7ea7419$exports, "removeDir", () => $c8f7705cf7ea7419$export$75426ec8836e4c8);
$parcel$export($c8f7705cf7ea7419$exports, "getFilePaths", () => $c8f7705cf7ea7419$export$9f499d9b98b35925);
$parcel$export($c8f7705cf7ea7419$exports, "copyFiles", () => $c8f7705cf7ea7419$export$6738b3634be42910);
$parcel$export($c8f7705cf7ea7419$exports, "fileExists", () => $c8f7705cf7ea7419$export$ffc21166d570a16);
$parcel$export($c8f7705cf7ea7419$exports, "readJson", () => $c8f7705cf7ea7419$export$6a368473377ea45c);
$parcel$export($c8f7705cf7ea7419$exports, "writeJson", () => $c8f7705cf7ea7419$export$476f628a921f7411);
$parcel$export($c8f7705cf7ea7419$exports, "readFile", () => $c8f7705cf7ea7419$export$72c04af63de9061a);
$parcel$export($c8f7705cf7ea7419$exports, "writeFile", () => $c8f7705cf7ea7419$export$552bfb764b5cd2b4);
$parcel$export($c8f7705cf7ea7419$exports, "runTransforms", () => $c8f7705cf7ea7419$export$c88707f9397b495c);



const $c8f7705cf7ea7419$export$2f93ca0bbcabd05f = async (directory)=>{
    try {
        await $7gN5q$fsextra.promises.access(directory, ($7gN5q$fsextra.constants || $7gN5q$fsextra).W_OK);
        return true;
    } catch (err) {
        return false;
    }
};
const $c8f7705cf7ea7419$export$ef2527cb59c03af5 = (root, options = {
    recursive: true
})=>$7gN5q$fsextra.mkdirSync(root, options)
;
const $c8f7705cf7ea7419$export$75426ec8836e4c8 = (path)=>$7gN5q$rimraf.sync(path)
;
const $c8f7705cf7ea7419$export$9f499d9b98b35925 = (patterns, options)=>$7gN5q$globby(patterns, options)
;
const $c8f7705cf7ea7419$export$6738b3634be42910 = (source, destination)=>$7gN5q$fsextra.copy(source, destination)
;
const $c8f7705cf7ea7419$export$ffc21166d570a16 = (path)=>$7gN5q$fsextra.existsSync(path)
;
const $c8f7705cf7ea7419$export$6a368473377ea45c = (...args)=>$7gN5q$fsextra.readJsonSync(...args)
;
const $c8f7705cf7ea7419$export$476f628a921f7411 = (...args)=>$7gN5q$fsextra.outputJSONSync(...args)
;
const $c8f7705cf7ea7419$export$72c04af63de9061a = (filePath)=>$7gN5q$fsextra.readFileSync(filePath, "utf8")
;
const $c8f7705cf7ea7419$export$552bfb764b5cd2b4 = (filePath, content, options = {
    force: false,
    append: false
})=>{
    const exists = $7gN5q$fsextra.existsSync(filePath);
    if (exists && options.append) return $7gN5q$fsextra.appendFileSync(filePath, content);
    if (!exists || exists && options.force) return $7gN5q$fsextra.outputFileSync(filePath, content);
    throw new Error(`Cannot write file to ${filePath}`);
};
const $c8f7705cf7ea7419$export$c88707f9397b495c = async (filePath, ...transforms)=>{
    const isStringPath = typeof filePath === "string";
    const sourcePath = isStringPath ? filePath : filePath.src;
    const destinationPath = isStringPath ? filePath : filePath.dest;
    let source = $7gN5q$fsextra.readFileSync(sourcePath, "utf8");
    for (const [transform, context] of transforms){
        const transformed = await transform(source, context);
        if (transformed) source = transformed;
    }
    if (destinationPath) return $7gN5q$fsextra.outputFile(destinationPath, source);
    return source;
};



var $fad6e5fbfd2318fb$exports = {};

$parcel$export($fad6e5fbfd2318fb$exports, "checkAndNotifyUpdates", () => $fad6e5fbfd2318fb$export$85481a6fc31b788c);


const $fad6e5fbfd2318fb$export$85481a6fc31b788c = async (packageJson)=>{
    try {
        const res = await $7gN5q$updatecheck(packageJson).catch(()=>null
        );
        if (res === null || res === void 0 ? void 0 : res.latest) {
            $62ae51800c28537f$export$af88d00dbe7f521.newLine();
            $62ae51800c28537f$export$af88d00dbe7f521.warning(`A new version of "${packageJson.name}" is available!`);
            $62ae51800c28537f$export$af88d00dbe7f521.log(`You can update by running: ${$62ae51800c28537f$export$af88d00dbe7f521.withVariable(`npm update ${packageJson.name}`)}`);
            $62ae51800c28537f$export$af88d00dbe7f521.newLine();
        }
        process.exit();
    } catch  {
    // ignore error
    }
};




export {$d2784300cb69549f$export$a48cded481a7b9a3 as filterDependencies, $d2784300cb69549f$export$883cc5315f212021 as filterTemplates, $f26fe0275fa4931b$export$b9c77372957d1829 as startDocker, $f26fe0275fa4931b$export$9238c7a74ca8f573 as stopDocker, $f26fe0275fa4931b$export$1fe1303d5ebdc741 as runMigrations, $f26fe0275fa4931b$export$85f637821bb9f000 as runPrismaCodegen, $f26fe0275fa4931b$export$d1a4d70edc68db32 as syncGeneratorMigrations, $e35cfb01f3b02780$export$49bc81080f205727 as prompts, $910700bd48d72431$export$84148c2dee5b43c5 as GeneratorLocation, $83f9063289dc0194$export$d31b835ca3ad9ed as filetypeToPrettierParser, $83f9063289dc0194$export$b0ee1b8c2e95dcbe as getPrettierParser, $83f9063289dc0194$export$fe3204bf5faa7e51 as transformPrettier, $de00a81b1a4c70e1$export$4a5dd64dec194a6a as transformEjs, $f3a3e1ac784d319c$export$259b2dad08caae9f as createJscodeshiftProgram, $f3a3e1ac784d319c$export$9b3cf6a05365ed8f as createJscodeshiftTransform, $17d5a96dac1073ea$export$d41ce78d713db213 as addImport, $8e7ad5eba6db344f$export$911865420682240f as findModuleExportsExpressions, $8235f45a91cc1e6c$export$cca4c4c5e6e9f10 as addPrismaEnum, $eedf22dd7a0eade6$export$8eb5c4001bc26e32 as addPrismaField, $919f5877cffaeae0$export$3af34c32d299916 as addPrismaGenerator, $87517b8de60d215a$export$a5156224d356ba0e as addPrismaModelAttribute, $8b3c464caa8bcb73$export$c9297bb316159832 as addPrismaModel, $6fd0cc407948fe8e$export$b0848f155f13a730 as produceSchema, $be99dba2948f5913$export$12acaf1194b381ec as setPrismaDataSource, $055006d47df8621a$export$de6b7d32e7276555 as logSchema, $1bdf442fb86e9adf$export$bbec2b7d78f06a8d as throwHandledError, $1bdf442fb86e9adf$export$255442a2e70017bf as onPromptCancel, $e68a37bf8acc13df$export$fa77af024d7e00a8 as execSync, $e68a37bf8acc13df$export$78e3044358792147 as exec, $e68a37bf8acc13df$export$e772c8ff12451969 as sleep, $c8f7705cf7ea7419$export$2f93ca0bbcabd05f as isWriteable, $c8f7705cf7ea7419$export$ef2527cb59c03af5 as makeDir, $c8f7705cf7ea7419$export$75426ec8836e4c8 as removeDir, $c8f7705cf7ea7419$export$9f499d9b98b35925 as getFilePaths, $c8f7705cf7ea7419$export$6738b3634be42910 as copyFiles, $c8f7705cf7ea7419$export$ffc21166d570a16 as fileExists, $c8f7705cf7ea7419$export$6a368473377ea45c as readJson, $c8f7705cf7ea7419$export$476f628a921f7411 as writeJson, $c8f7705cf7ea7419$export$72c04af63de9061a as readFile, $c8f7705cf7ea7419$export$552bfb764b5cd2b4 as writeFile, $c8f7705cf7ea7419$export$c88707f9397b495c as runTransforms, $62ae51800c28537f$export$9852986a3ec5f6a0 as table, $62ae51800c28537f$export$a0b6f61a94abd245 as chalk, $62ae51800c28537f$export$af88d00dbe7f521 as logger, $fad6e5fbfd2318fb$export$85481a6fc31b788c as checkAndNotifyUpdates};
